src/tests/ftest/container/query_attribute.py

```diff
@@ -0,0 +1,110 @@
+        name and corresponding value, so set unique name-value pair for each of
+        the 5 attributes.
+        2. Call continaer list-attrs. This returns the 5 attribute names, so
+        compare them against the actual names used.
+        3. Call container get-attr. This returns the value of the corresponding
+        name, so do this for each of the 5 attributes.
+
+    :avocado: recursive
+    """
+
+    def test_container_query_attr(self):
+        """JIRA ID: DAOS-4640
+
+        Test Description:
+            Test daos container query and attribute commands as described
+            above.
+
+        Use Cases:
+            Test container query, set-attr, list-attr, and get-attr commands.
+
+        :avocado: tags=all,pool,small,full_regression,cont_query_attr
+        """
+        # 1. Test pool query.
+        kwargs = {"scm_size": "1G"}
+        pool_create_result = self.get_dmg_command().get_output(
+            "pool_create", **kwargs)
+        expected_pool_uuid = pool_create_result[0]
+        sr = pool_create_result[1]
+        daos_cmd = DaosCommand(self.bin)
+        # Create container and store the UUID as expected.
+        kwargs = {"pool": expected_pool_uuid, "svc": sr}
+        expected_cont_uuid = daos_cmd.get_output(
+            "container_create", **kwargs)[0]
+        # Call daos container query, obtain pool and container UUID, and
+        # compare against those used when creating the pool and the container.
+        kwargs["cont"] = expected_cont_uuid
+        query_output = daos_cmd.get_output("container_query", **kwargs)[0]
+        actual_pool_uuid = query_output[0]
+        actual_cont_uuid = query_output[1]
+        self.assertEqual(actual_pool_uuid, expected_pool_uuid)
+        self.assertEqual(actual_cont_uuid, expected_cont_uuid)
+
+        # 2. Test container set-attr, get-attr, and list-attrs.
+        expected_attrs = []
+        expected_attrs_dict = {}
+        sample_attrs = []
+        sample_vals = []
+        # Create 5 attributes.
+        for i in range(5):
+            sample_attr = "attr" + str(i)
+            sample_val = "val" + str(i)
+            sample_attrs.append(sample_attr)
+            sample_vals.append(sample_val)
+            _ = daos_cmd.container_set_attr(
+                pool=actual_pool_uuid, cont=actual_cont_uuid,
+                attr=sample_attr, val=sample_val, svc=sr).stdout
+            expected_attrs.append(sample_attr)
+            expected_attrs_dict[sample_attr] = sample_val
+        expected_attrs.sort()
+        # List the attribute names.
+        kwargs = {
+            "pool": actual_pool_uuid,
+            "svc": sr,
+            "cont": actual_cont_uuid
+        }
+        actual_attrs = daos_cmd.get_output("container_list_attrs", **kwargs)
+        actual_attrs.sort()
+        self.assertEqual(actual_attrs, expected_attrs)
+        # Verify each attribute's value.
+        for i in range(5):
```

Comment:
I might consider doing something like:
```
errors = []
for i in range(5):
    kwargs["attr"] = sample_attrs[i]
    actual_val = daos_cmd.get_output("container_get_attr", **kwargs)[0]
    if sample_vals[i] != actual_val:
        errors.append("{} != {}".format(sample_vals[i], actual_val))
self.assertEqual(len(errors), 0, '\n'.join(errors))
```

This way it will finish looping through all the `sample_attrs` before failing the test.

Reply:
Thanks for suggesting. I implemented it.